### PR TITLE
tool to generate insertSigned() parameters RCHAIN-2830

### PR DESCRIPTION
**Pull request by dckc**
## Overview
The technique to sign over an unforgeable name for insertSigned (e.g. in MakeMint.rho) is insufficiently documented. This code automates the task and provides documentation of one form.

Example run:

```
sbt:casper> console
Welcome to Scala 2.12.8 (OpenJDK 64-Bit Server VM, Java 11.0.1).

scala> import coop.rchain.casper.util.rholang._
import coop.rchain.casper.util.rholang._

scala> RegistrySigGen.main(Array())

       given     1. sk = cd37277541a6d9ec4e635d895ef99e98961d7681e81e1fe59e321f8fdb1ee769
       given     2. timestamp = 1548819827686
       lastNonce 3. nonce = 9223372036854775807
 1,    ed25519   4. pk = 400dec5ede53008797c5f5b223dda748178e32a7c6095e75e3776776c45d8ea2
 4, 2, genIds    5. uname = Unforgeable(0xb1198287e706768ab01dcfc5d7c7aa65b832f6267c95052638ae95b3c9d02138)
 3, 5, registry  6. value = (9223372036854775807, bundle+ {   Unforgeable(0xb1198287e706768ab01dcfc5d7c7aa65b832f6267c95052638ae95b3c9d02138) })
 6,    protobuf  7. toSign = 2a3eaa013b0a0d2a0b10feffffffffffffffff010a2a5a280a243a220a20b1198287e706768ab01dcfc5d7c7aa65b832f6267c95052638ae95b3c9d021381001
 7, 1, ed25519   8. sig = de6369e0e4365a99acb7f7c3a82090293f7538d8872cf7b551db79fd3e20f36209dcf97a42225c717fa2c7df2c6de594baca67946c70bf92a7f72b66278d9d02

 
 rs!(
   "400dec5ede53008797c5f5b223dda748178e32a7c6095e75e3776776c45d8ea2".hexToBytes(),
   (9223372036854775807, bundle+{*CONTRACT}),
   "de6369e0e4365a99acb7f7c3a82090293f7538d8872cf7b551db79fd3e20f36209dcf97a42225c717fa2c7df2c6de594baca67946c70bf92a7f72b66278d9d02".hexToBytes(),
   *uriOut)
```

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-2830

### Notes
I checked that it produces the same results as in `MakeMint.rho`.

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [ ] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### CODE REVIEW

File: casper/src/test/scala/coop/rchain/casper/util/rholang/RegistrySigGen.scala [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  def insertSignedTerm(key: (Array[Byte], Array[Byte]), timestamp: Long, contract: String) = {
+    val (secKey, pubKey) = key
+    val user             = byteArrayToByteString(pubKey)
+
+    val seed        = DeployData().withUser(user).withTimestamp(timestamp)
+    val rand        = Blake2b512Random(DeployData.toByteArray(seed))
+    val id          = ByteString.copyFrom(rand.next())
+    val uname: Par  = GPrivate(id)
+    val nonce       = maxint
+    val toSign: Par = ETuple(Seq(GInt(nonce), uname))
```
Comment (dckc) [inline casper/src/test/scala/coop/rchain/casper/util/rholang/RegistrySigGen.scala]: oops; forgot the bundle around uname

File: casper/src/test/scala/coop/rchain/casper/util/rholang/RegistrySigGen.scala [hunk 2]
```diff
@@ -0,0 +1,107 @@
+package coop.rchain.casper.util.rholang
```
Comment (odeac) [inline casper/src/test/scala/coop/rchain/casper/util/rholang/RegistrySigGen.scala]: I think a better location is casper/src/main/...

File: casper/src/test/scala/coop/rchain/casper/util/rholang/RegistrySigGen.scala [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+import coop.rchain.crypto.codec.Base16
+import coop.rchain.crypto.hash.Blake2b512Random
+import coop.rchain.crypto.signatures.{Ed25519, Secp256k1}
+import coop.rchain.models.Expr.ExprInstance.GInt
+import coop.rchain.models.Expr.ExprInstance.{GBool, GByteArray, GString}
+import coop.rchain.models.rholang.implicits._
+import coop.rchain.models.{Bundle, ETuple, Expr, GPrivate, Par}
+import coop.rchain.rholang.interpreter.{PrettyPrinter, Runtime}
+
+case class InsertSigned(pk: Hex, value: (Long, Contract), sig: Hex) {
```
Comment (odeac) [inline casper/src/test/scala/coop/rchain/casper/util/rholang/RegistrySigGen.scala]: wart remover will complain about non-final case classes

Comment (dckc) [reply casper/src/test/scala/coop/rchain/casper/util/rholang/RegistrySigGen.scala]: is it complaining for a good reason? Or should I add an annotation to suppress the warning?

Comment (dckc) [reply casper/src/test/scala/coop/rchain/casper/util/rholang/RegistrySigGen.scala]: oh... it doesn't mind that I'm overriding toString... just that I'm missing `final`. Fair enough.

**Comment by dckc** [comment]
all set now, @odeac ?

**Pull request merged**